Sources/Sentry/SentryTraceState.m

```diff
@@ -0,0 +1,117 @@
+#import "SentryTraceState.h"
+#import "SentryDsn.h"
+#import "SentryLog.h"
+#import "SentryOptions+Private.h"
+#import "SentryScope+Private.h"
+#import "SentrySerialization.h"
+#import "SentryTracer.h"
+#import "SentryUser.h"
+
+@implementation SentryTraceStateUser
+
+- (instancetype)initWithUserId:(NSString *)userId segment:(NSString *)segment
+{
+    if (self = [super init]) {
+        _userId = userId;
+        _segment = segment;
+    }
+    return self;
+}
+
+- (instancetype)initWithUser:(nullable SentryUser *)user
+{
+    return [self initWithUserId:user.userId segment:[user.data objectForKey:@"segment"]];
```

Comment:
`h`: Can we be sure that `[user.data objectForKey:@"segment"]` returns a NSString? I think we need to check that.

Reply:
`m`: User can be `nil`. I think it would be nice to check for that.

Reply:
Well, to be honest, users can add anything in the dictionary.
Should we check only for string? 
Try to serialize if is another type of object?

Reply:
We use the `userObject` from the scope so it can be basically anything. Segment is an `NSString` so of course we need to check that.